persistence/album_repository.go

```diff
@@ -79,6 +87,14 @@ func NewAlbumRepository(ctx context.Context, db dbx.Builder) model.AlbumReposito
 		}
 	}
 
+	if conf.Server.AlbumPlaycountMode == "normalized" {
+		r.AddHook(func(album *model.Album) {
+			if album.SongCount != 0 {
+				album.PlayCount = album.PlayCount / int64(album.SongCount)
+			}
+		})
+	}
```

Comment:
I may be missing something, but why adding a hook system and adding a hook to calculate this later, instead of just calculating and setting the value here?

```suggestion
	if conf.Server.AlbumPlaycountMode == "normalized" {
		if album.SongCount != 0 {
			album.PlayCount = album.PlayCount / int64(album.SongCount)
		}
	}
```

Reply:
Because future plugins may want to make similar changes when parsing albums. Having a hook system in place should make that easier and more maintainable. It is debatable whether this is already required when there is only one such simple "plugin" though.

Note that the hook is installed in `NewAlbumRepository()` but invoked in `toModels()`. If you don't want to use a hook, we have to move the check + change into that function. It feels better to keep this function small and not have feature-specific code in that function though.

Reply:
The thing is, PostScan is already a "hook", the check can be done inside it, as it is meant to map what is in the DB to our model (domain).

This "hook system" can definitely be useful in more complex scenarios, but I think this is an overkill design for the problem at hand.

And, if we want it to be really useful, I think it should be part of the model package, not the persistence one. But as I said, I think this is overkill for now.

Reply:
I moved the functionality to `toModels()`. Please check if this is what we want or whether we should look at PostScan or switching to the model package instead.